### PR TITLE
Variable metadata

### LINKED ISSUE #126: Add hover text definitions for each column in listing.
Heather requested this 12/17/18 and is working on creating definitions for each of the columns in the listing/ She wants these to display when the user hovers over the column header

# Test notes
* Listing column headers have tooltips.

### PR DESCRIPTION
**Pull request by samussiah**
# Issues
Closes #126

### CODE REVIEW

File: src/listing/onInit/applyVariableMetadata.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            description: 'Site response to the query',
+        },
+    ];
+
+    this.config.descriptions = [];
+    this.config.cols.forEach((col,i) => {
+        const md = variableMetadata
+            .find(variableMetadatum => variableMetadatum.variable === col);
+        this.config.headers[i] = md ? md.label : col;
+        this.config.descriptions.push(md ? md.description : col);
```
Comment (pburnsdata) [inline src/listing/onInit/applyVariableMetadata.js:122]: good safeguards as always!

File: src/listing/onDraw/truncateCellText.js [hunk 2]
```diff
@@ -3,6 +3,6 @@ export default function truncateCellText() {
         this.tbody
             .selectAll('td')
             .attr('title', d => d.text)
-            .filter(d => d.text.length > this.chart.initialSettings.truncation_cutoff)
-            .text(d => `${d.text.substring(0, this.chart.initialSettings.truncation_cutoff)}...`);
```
Comment (pburnsdata) [inline src/listing/onDraw/truncateCellText.js:7]: good call on that truncation doeee

File: src/listing/onInit/applyVariableMetadata.js [hunk 3]
```diff
@@ -3,101 +3,122 @@ export default function applyVariableMetadata() {
         {
             variable: this.initialSettings.site_col,
             label: 'Site',
-            description: 'Site name assigned in EDC system',
+            description: 'Site name assigned in EDC system'
```
Comment (pburnsdata) [inline src/listing/onInit/applyVariableMetadata.js:6]: good catch

**Review by pburnsdata — APPROVED** [review]
Perfecto

**Review by danedexF5 — APPROVED** [review]

**Pull request merged**